openlibrary/catalog/add_book/__init__.py

```diff
@@ -646,21 +654,39 @@ def load_data(rec, account_key=None):
         del rec['cover']
     try:
         # get an OL style edition dict
-        edition = build_query(rec)
+        rec_as_edition = build_query(rec)
+        edition: dict[str, Any]
+        if existing_edition:
+            edition = existing_edition.dict() | rec_as_edition
```

Comment:
```suggestion
            # Note: This will overwrite any fields in the existing edition. This is ok for
            # now, because we'll really only come here when overwriting a promise
            # item
            edition = existing_edition.dict() | rec_as_edition
```